### ISSUE TITLE
Points are drawn twice in openGL

### ISSUE THREAD

**Issue by clankill3r** [processing/processing#5344]
Points are drawn twice in openGL.

```java

void setup() {
  size(800, 800, P3D);
}

void draw() {
  background(0);
  stroke(255, 0, 0);
  point(width/2, height/2);
}
```

<img width="90" alt="screen shot 2017-12-23 at 21 04 37" src="https://user-images.githubusercontent.com/738650/34322352-13c2afa6-e825-11e7-9ddc-f4d36b6f3733.png">

Processing 3.3.6
macOS 10.12.6
retina screen

**Comment by jeremydouglass** [comment]
Huh. I'm pretty sure that second pixel is just an expected rendering artifact of a 3D point...?

By comparison, look at the 1x1 through 5x5 rects().

If you need to draw a clean single pixel on top of a 3D canvas, you could just use set(). That sets a single pixel. 

```
void setup() {
  size(100, 100, P3D);
}

void draw() {
  background(0);
  stroke(255, 0, 0);
  set(10, height/2, color(255,0,0));
  point(20, height/2);
  rect(30, height/2, 1, 1);
  rect(40, height/2, 2, 2);
  rect(50, height/2, 3, 3);
  rect(60, height/2, 4, 4);
  rect(70, height/2, 5, 5);
}

```

**Comment by clankill3r** [comment]
The same thing happens in 2D.

**Comment by JakubValtar** [comment]
This is antialiasing acting up. The point has center at the pixel corner, so your point contributes samples to more pixels. I just wonder why it is not 2x2 square like on windows (macOS probably uses different method of aa sampling).

If you want your point to be a crisp single pixel
- draw it on a pixel center: `point(width/2+0.5, height/2+0.5)`
- or use P2D and square points (they will get aligned to the pixel grid): `strokeCap(SQUARE)`

In short: P2D aligns 1px lines and square points to the grid, in all other cases you have to align yourself (+0.5), because whole coordinates are between the pixels.
